Create policy-rc.d in target to prevent daemon starting

If an optional /usr/sbin/policy-rc.d script exists, it will be run by
invoke-rc.d before deciding to start daemons. By exiting with code 101,
any actions are prevented from running. This effectively keeps daemons
from being started in the chroot and is needed when systemd is init
since the start-stop-daemon hack doesn't work in that case.

This is borrowed from the upstream sid debootstrap script, which took it
from debian-installer.

http://anonscm.debian.org/cgit/d-i/debootstrap.git/commit/?id=29cd6bee

--- a/obs-debootstrap-script
+++ b/obs-debootstrap-script
@@ -133,6 +133,11 @@ second_stage_install () {
 	sed -i '/dpkg.*--compare-versions/ s/\<lt\>/lt-nl/' "$TARGET/var/lib/dpkg/info/initscripts.postinst"
 	fi
 
+	echo \
+"#!/bin/sh
+exit 101" > "$TARGET/usr/sbin/policy-rc.d"
+	chmod 755 "$TARGET/usr/sbin/policy-rc.d"
+        
 	mv "$TARGET/sbin/start-stop-daemon" "$TARGET/sbin/start-stop-daemon.REAL"
 	echo \
 "#!/bin/sh
@@ -191,6 +196,7 @@ echo \"Warning: Fake initctl called, doi
 		mv "$TARGET/sbin/initctl.REAL" "$TARGET/sbin/initctl"
 	fi
 	mv "$TARGET/sbin/start-stop-daemon.REAL" "$TARGET/sbin/start-stop-daemon"
+	rm -f "$TARGET/usr/sbin/policy-rc.d"
 
 	progress $bases $bases CONFBASE "Configuring base system"
 	info BASESUCCESS "Base system installed successfully."
